Assets/MixedRealityToolkit.Services/InputSimulation/SimulatedGestureHand.cs

```diff
@@ -148,40 +149,47 @@ protected override void UpdateInteractions(SimulatedHandData handData)
                             {
                                 InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction);
 
-                                // Stop gestures
-                                CompleteHoldGesture();
-                                CompleteManipulationNavigationGesture();
+                                // Stop active gestures
+                                CompleteHold();
+                                CompleteManipulation();
+                                CompleteNavigation();
                             }
                         }
                         else if (Interactions[i].BoolData)
                         {
-                            if (!manipulationInProgress)
+                            if (manipulationInProgress)
+                            {
+                                UpdateManipulation();
+                            }
+
+                            if (navigationInProgress)
+                            {
+                                UpdateNavigation();
+                            }
```

Comment:
Also when we're done with this PR, it might be worth updating the input docs to specify the expected order (basically copy paste what you had in the issue into the docs)

Reply:
I've added a section to the documentation about the expected order of gesture events.

I've also linked the Gesture Tester example script there, which has proven useful to devs in the past for correct usage of the events, particularly for distinguishing the 3 event types based on generic InputEventData.

Reply:
Looks great, thanks!